calabash-cucumber/lib/calabash-cucumber/launch/simulator_launcher.rb

```diff
@@ -251,13 +251,24 @@ def app_bundle_or_raise(path=nil, device_build_dir='iPhoneSimulator')
             puts('-'*37)
           end
         else
-          bo_dir = build_output_dir_for_project
-          sim_dirs = Dir.glob(File.join(bo_dir, '*-iphonesimulator', '*.app'))
+          search_dir = build_output_dir_for_project || DERIVED_DATA
+          sim_dirs = ''
+          apps = `find #{search_dir} -name "*.app" | sort -n`.split("\n")
+          apps.each do |app_path|
+            if app_path.split("/")[-2].include? 'iphonesimulator'
+              app =  RunLoop::App.new(app_path)
+              executable_name = app.executable_name
+              path_to_bin = app_path + "/" + executable_name
+              if `xcrun strings "#{path_to_bin}" | grep -E 'CALABASH VERSION'`.include? "CALABASH VERSION"
```

Comment:
I don't want to make more work for you, but do you think you could implement this in:

* [run-loop/lib/ipa.rb](https://github.com/calabash/run_loop/blob/develop/lib/run_loop/ipa.rb)
* [run-loop/lib/app.rb](https://github.com/calabash/run_loop/blob/develop/lib/run_loop/app.rb)

The method name in each should be `calabash_server_version`.  You can duplicate the code in each - you don't have to worry about repeating yourself.  If you are really ambitious you can write a `Strings` class that can call strings on an arbitrary file.

There is a small problem with just querying the executable for the server - sometimes apps dynamically link calabash.  This is different from dylib injection.  In this case, the app _loads a calabash dylib from the app bundle at runtime_.

I am not overly worried about his case.  User that do use this kind of dynamic linking will probably be savvy enough to set the `APP` variable.